cordapp-cpk/src/test/resources/external-cordapp/build.gradle

```diff
@@ -0,0 +1,40 @@
+plugins {
+    id 'net.corda.plugins.cordapp-cpk'
+    id 'maven-publish'
+}
+
+apply from: 'repositories.gradle'
+apply from: 'javaTarget.gradle'
+
+group = 'com.example'
+version = cordapp_version
+
+cordapp {
+    targetPlatformVersion = platform_version.toInteger()
+    minimumPlatformVersion = platform_version.toInteger()
+
+    contract {
+        name = 'External CorDapp'
+        versionId = cordapp_contract_version.toInteger()
+        licence = 'Test-Licence'
+        vendor = 'R3'
+    }
+}
+
+jar {
+    archiveBaseName = 'external-cordapp'
+}
+
+publishing {
+    repositories {
+        maven {
+            name = 'Test'
+            url = getProperty("maven.repository.dir")
+        }
+    }
+    publications {
+        maven(MavenPublication) {
+            artifact cpk
```

Comment:
what do you mean with it being "incomplete"? what is the issue with only publishing the cpk (which is all I need here)?

Reply:
> what do you mean with it being "incomplete"? what is the issue with only publishing the cpk (which is all I need here)?

The CPK has a `cordapp` classifier, and so your published Maven artifact has no "default" artifact, i.e. no "main" jar. This in turn means that the POM's `<packaging>` type will be `pom` and not `jar`, and _that_ means that the `cordapp-cpk` plugin will not also publish the "companion" POM artifact that makes transitive CPK dependencies work...

Reply:
I see... it probably works simply because external-cordapps doesn't have any other CPK dependency itself, may I should actually add one to test this as well?

Reply:
> I see... it probably works simply because `external-cordapps` doesn't have any other CPK dependency itself, maybe I should actually add one to test this as well?

Yes, something like that. Actually even as little as adding
```gradle
dependencies {
    cordaProvided project(':corda-api')
}
```
to `external-cordapp` would do because the "companion" POM contains both `cordaProvided` and `cordapp` dependencies. You'd just need to publish the `corda-api` jar to your "Test" repository too.

Reply:
but if I declare `corda-api` as cordaProvided it won't be included in the CPB, how would that test be meaningful?

Reply:
> but if I declare `corda-api` as `cordaProvided` it won't be included in the CPB, how would that test be meaningful?

Hmm, OK. I was thinking that transitively resolving the `corda-api` artifact would allow your `cordapp-cpb` project to depend on `corda-api`, and so allow it to include a stub contract class etc. _However_, that's probably testing the `cordapp-cpk` plugin rather than the `cordapp-cpb` plugin :grin:.

If you want to resolve transitive CPKs 3 levels deep then you'll need something that looks more like `transitive-cordapps`. Have you also tested that "system CPKs" are _not_ packaged?

Reply:
> If you want to resolve transitive CPKs 3 levels deep then you'll need something that looks more like transitive-cordapps. Have you also tested that "system CPKs" are not packaged?

I've tested it manually but it would be good to have a unit test instead